src/DbExportsRegistryController.cpp

```diff
@@ -0,0 +1,137 @@
+#include "DbExportsRegistryController.h"
+#include "PromptWidget.h"
+
+#include <QFileDialog>
+#include <QMessageBox>
+#include <QSettings>
+
+DbExportsRegistryController::DbExportsRegistryController(const QString &settingsFilePath, QObject *parent) :
+    QObject(parent), isExportRequestMessageVisible(false), window(nullptr)
+{
+    dbExportsRegistry = new DbExportsRegistry(settingsFilePath, this);
+    connect(dbExportsRegistry, &DbExportsRegistry::dbExportRecommended, this, &DbExportsRegistryController::handleDbExportRecommended);
+}
+
+DbExportsRegistryController::~DbExportsRegistryController()
+{
+    dbExportsRegistry->deleteLater();
+}
+
+void DbExportsRegistryController::setMainWindow(MainWindow *window)
+{
+    DbExportsRegistryController::window = window;
+    if (wasDbExportRecommendedWithoutMainWindow)
+        handleDbExportRecommended();
+
+    wasDbExportRecommendedWithoutMainWindow = false;
+}
+
+void DbExportsRegistryController::setWSClient(WSClient *wsClient)
+{
+    DbExportsRegistryController::wsClient = wsClient;
+    connect(wsClient, &WSClient::cardDbMetadataChanged, this, &DbExportsRegistryController::handleCardIdChanged);
+    connect(wsClient, &WSClient::statusChanged, this, &DbExportsRegistryController::handleDeviceStatusChanged);
+    connect(wsClient, &WSClient::connectedChanged, this, &DbExportsRegistryController::handleDeviceConnectedChanged);
+
+    dbExportsRegistry->setCurrentCardId(wsClient->get_cardId());
```

Comment:
are you getting the CPZ directly here, or its hash?

Reply:
As I remember the card id is made from a hash of the cpz. I'll check it any way.